### PR TITLE
fix: camera permission alert with settings access

### LINKED ISSUE #10642: [Bug]: Camera Permission Denial Leads to Poor User Experience with No Easy Way to Fix
### Describe the bug

During the initial wallet setup in the MetaMask mobile app, users encounter an issue related to camera permissions that can lead to a suboptimal user experience:

1. Initial State: When users first set up their wallet, a QR scanner icon is available in the top right corner.

2. Camera Permission Request: Upon tapping the QR scanner icon, users receive a system alert: "MetaMask Would like to Access the Camera"
 
3. Permission Denial Scenario: If the user selects "Don't Allow":
a. The app doesn't provide an easy way for users to later opt back into camera permissions.
b. Subsequent attempts to use the QR scanner result in a non-helpful alert, leaving users without clear guidance on how to enable the camera. Unhelpful alert shown below

image

Impact:
This issue significantly hampers the user experience, particularly for those who initially or accidentally deny camera access but later wish to use the QR scanning functionality. The lack of a clear path to re-enable permissions may lead to user frustration and reduced app functionality.

Solution:
Enable the user to opt back in with a more useful alert message when tapping on the QR Code icon. The alert below tells the user how they can fix it and allows the user to be routed to their system settings to opt-back in

Example Alert:
image

### Expected behavior

Expected alert that allows user to be linked into their system settings where they can easily opt back in camera permissions

### Screenshots/Recordings

camera_permissions

### Steps to reproduce

1. Fresh install the MetaMask mobile app.
2. Complete the initial wallet setup.
3. Tap the QR scanner icon in the top right corner
5. When prompted "MetaMask Would like to Access the Camera", select "Don't Allow".
6. Attempt to use the QR scanner again by tapping the icon.
7. Observe the non-helpful alert that appears.

### Error messages or log output

_No response_

### Detection stage

In production (default)

### Version

7.28.0

### Build type

None

### Device

iPhone 12 Pro Max

### Operating system

iOS

### Additional context

NA

### Severity

- Poor user experience

### PR DESCRIPTION
**Pull request by vinnyhoward**
## **Description**

This pull request addresses an issue related to camera permissions during the initial wallet setup in the app. The improvement provides a better user experience for those who initially deny camera access but later wish to use the QR scanning functionality.

implemented a solution to improve the user experience when dealing with camera permissions:

1. Replaced the old alert with a new, informative message when users attempt to use the QR scanner after initially denying permission.

2. Added functionality to route users directly to their system settings to modify camera permissions.

3. Enhanced user experience by providing a clear path to enable necessary app functionality.

## **Related issues**

Fixes: [Issue #10642 ](https://github.com/MetaMask/metamask-mobile/issues/10642)

## **Manual testing steps**

1. Fresh install the MetaMask mobile app.
2. Complete the initial wallet setup.
3. Tap the QR scanner icon in the top right corner.
4. When prompted "MetaMask Would Like to Access the Camera", select "Don't Allow".
5. Attempt to use the QR scanner again by tapping the icon.
6. Verify that the new alert appears with options to cancel or open settings.
7. Test that the "Open Settings" option correctly routes to the system settings.
8. Enable camera permissions in settings and return to the app.
9. Confirm that the QR scanner now functions correctly.

## **Screenshots/Recordings**
iOS
| Before  | After  |
|:---:|:---:|
| before|after| 

Android
| Before  | After  |
|:---:|:---:|
|before_android_compressed |after_android_compressed| 

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by yanrong-chen** [comment]
Updated copy here 
Screenshot 2024-08-16 at 10 47 50 AM

**Review by EtherWizard33 — APPROVED** [review]
nice, approved!

**Review by georgewrmarshall — APPROVED** [review]
LGTM! PR description is a thing of beauty! One non-blocking comment to check with our content design team.
- Checked manual testing steps locally ✅ 

I noticed that once we have completed the steps the simulator will still not open the camera but I'm assuming that is expected behavior of the simulator


https://github.com/user-attachments/assets/f9782388-b4af-4cb1-a6eb-1d74aad720e7

File: locales/languages/en.json [hunk 1]
```diff
@@ -1348,13 +1348,14 @@
     "error": "Error",
     "attempting_to_scan_with_wallet_locked": "Looks like you're trying to scan a QR code, you need to unlock your wallet to be able to use it.",
     "attempting_sync_from_wallet_error": "Looks like you're trying to sync with the extension. In order to do so, you will need to erase your current wallet. \n\nOnce you've erased or reinstalled a fresh version of the app, select the option to \"Sync with MetaMask Extension\". Important! Before erasing your wallet, make sure you've backed up your Secret Recovery Phrase.",
-    "not_allowed_error_title": "Enable camera access",
-    "not_allowed_error_desc": "Camera access needed to scan a QR code associated with an account address or a contract address",
+    "not_allowed_error_title": "Turn on camera access",
+    "not_allowed_error_desc": "To scan a QR code, you'll need to give MetaMask camera access from your device's settings menu.",
```
[the remark this answers is not in the document]
Comment (coreyjanssen) [reply locales/languages/en.json:1352]: Thanks for asking @georgewrmarshall! In this case, we should have a period as the body copy is quite long and uses a comma.

[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply locales/languages/en.json:1352]: Thanks @coreyjanssen 🙏

**Pull request merged**